dialog/file.go

```diff
@@ -181,10 +184,36 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 
 	favorites := f.loadFavorites()
 
-	favoritesGroup := widget.NewGroup("Favorites", favorites...)
-	return fyne.NewContainerWithLayout(layout.NewBorderLayout(header, footer, favoritesGroup, nil),
-		favoritesGroup, header, footer, body)
+	favoritesGroup := widget.NewVScrollContainer(widget.NewGroup("Favorites", favorites...))
+	var optionsButton *widget.Button
+	optionsButton = widget.NewButton("Options", func() {
+		f.optionsMenu(fyne.CurrentApp().Driver().AbsolutePositionForObject(optionsButton), optionsButton.Size())
+	})
+
+	left := fyne.NewContainerWithLayout(layout.NewBorderLayout(nil, optionsButton, nil, nil), favoritesGroup, optionsButton)
+
+	return fyne.NewContainerWithLayout(layout.NewBorderLayout(header, footer, left, nil),
+		left, header, footer, body)
+
+}
+
+func (f *fileDialog) optionsMenu(position fyne.Position, buttonSize fyne.Size) {
+	hiddenFiles := widget.NewCheck("Show Hidden Files", func(changed bool) {
+		f.showHidden = changed
+		f.refreshDir(f.dir)
+	})
+	hiddenFiles.SetChecked(f.showHidden)
+
+	title := widget.NewLabel("Options")
```

Comment:
I don’t think this options label is needed. The user would have pressed a button with the same name to bring up the menu. I think it can safely be removed from the menu.

Reply:
A matter of taste I guess, any other views? Here it is without the title
![image](https://user-images.githubusercontent.com/20552436/103176032-8fb80d00-4866-11eb-99d8-815b9f10d624.png)

Reply:
Thanks for sharing a picture with it removed. I certainly prefer it that way. With the title label, it mostly looks like another options button to press, IMHO.

Reply:
Perhaps if the menu was above or below, instead of to the right of, the button the header is less needed?

Reply:
Or options could be an Accordion

Reply:
> Doesn't an accordion have to have space to expand into?

Sure, but when Options expands, Favourites collapses so there would be space available

> Isn't a Popup cleaner?

Maybe, we'll only know if explore the different options and compare the results :)

Reply:
I think that would have to go hand in hand with a complete overhaul. Is what I have made acceptable for release?